### PR TITLE
fix(nuxt): pass `joinRelativeURL` to bundle renderer and share paths with nitro

### PR DESCRIPTION
**Pull request by danielroe**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Previously we switched across to `joinRelativeURL` in `build/paths.mjs` but didn't do so for `paths.ts` which is used in the nitro renderer.

This PR:
1. updates the implementations to be the same
2. deduplicates these two files on the server when building

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have added tests (if possible).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxt/src/core/runtime/nitro/paths.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { joinURL } from 'ufo'
+import { joinRelativeURL as joinURL } from 'ufo'
```
Comment (pi0) [inline packages/nuxt/src/core/runtime/nitro/paths.ts:1]: I would suggest using `joinRelativeURL` as the canonical name to make today's (and future) tracing easier.

File: packages/nuxt/src/app/entry.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import type { App } from 'vue'
 
 // These files must be imported first as they have side effects:
 // 1. (we set __webpack_public_path via this import, if using webpack builder)
-import '#build/paths.mjs'
+import '#paths'
```
Comment (pi0) [inline packages/nuxt/src/app/entry.ts:6]: i'm sure nitro at least will keep using `#internal/nitro/` scoped virtuas but it _might_ conflict with any other alias by modules we could use `#nuxt-paths` or something non conflicting. totally up to you.

File: vitest.config.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ export default defineConfig({
   resolve: {
     alias: {
       '#build/nuxt.config.mjs': resolve('./test/mocks/nuxt-config'),
-      '#build/paths.mjs': resolve('./test/mocks/paths'),
+      '#paths': resolve('./test/mocks/paths'),
```
Comment (pi0) [inline vitest.config.ts:11]: i'm sure nitro at least will keep using `#internal/nitro/` scoped virtuas but it _might_ conflict with any other alias by modules we could use `#nuxt-paths`, `#app-paths`, `#app/paths` or something non conflicting. totally up to you.

Comment (danielroe) [reply vitest.config.ts:11]: maybe `#internal/nuxt/paths`? what do you think?

**Review by pi0 — APPROVED** [review]
🆒 💯

**Pull request merged**